annubes/task.py

```diff
@@ -330,25 +332,28 @@ def _check_int_positive(self, name: str, value: Any, strict: bool) -> None:  # n
             msg = f"`{name}` must be greater than or equal to 0."
             raise ValueError(msg)
 
-    def _check_bool(self, name: str, value: Any) -> None:  # noqa: ANN401
-        if not isinstance(value, bool):
-            msg = f"`{name}` must be a boolean."
-            raise TypeError(msg)
+    def _check_range(self, name: str, value: Any) -> None:  # noqa: ANN401
+        if isinstance(value, tuple):
+            for v in value:
+                self._check_int_positive(name, v, strict=False)
+        else:
+            self._check_int_positive(name, value, strict=False)
 
     def _check_time_vars(self) -> None:
         strictly_positive = {
             "stim_time": (self.stim_time, True),
             "dt": (self.dt, True),
             "tau": (self.tau, True),
-            "fix_time": (self.fix_time, False),
         }
         for name, value in strictly_positive.items():
             self._check_int_positive(name, value[0], strict=value[1])
```

Comment:
now that all are `True` for strictness, we can simplify like this:

```suggestion
        strictly_positive = {
            "stim_time": self.stim_time,
            "dt": self.dt,
            "tau": self.tau,
        }
        for name, value in strictly_positive.items():
            self._check_int_positive(name, value, strict=True)
```